src/transformers/custom-elements-manifest/get-declarations.ts

```diff
@@ -0,0 +1,334 @@
+ * @param declaration
+ * @param context
+ */
+function* getClassFieldsForDeclaration(declaration: ComponentDeclaration, context: TransformerContext): IterableIterator<schema.ClassField> {
+	const visibility = context.config.visibility ?? "public";
+	for (const member of declaration.members) {
+		if (member.visibility === visibility && member.propName != null) {
+			yield {
+				kind: "field",
+				name: member.propName,
+				privacy: member.visibility,
+				description: member.jsDoc?.description,
+				type: typeToSchemaType(context, member.typeHint || member.type?.()),
+				default: member.default != null ? JSON.stringify(member.default) : undefined,
+				inheritedFrom: getInheritedFromReference(declaration, member, context),
+				summary: getSummaryFromJsDoc(member.jsDoc)
+				// TODO: "static"
+			};
+		}
+	}
+}
+
+/**
+ * Returns method docs for a declaration
+ * @param declaration
+ * @param context
+ */
+function* getMethodsForDeclaration(declaration: ComponentDeclaration, context: TransformerContext): IterableIterator<schema.ClassMethod> {
+	const visibility = context.config.visibility ?? "public";
+	for (const method of declaration.methods) {
+		const parameters: schema.Parameter[] = [];
+		const node = method.node;
+		let returnType: tsModule.Type | undefined = undefined;
+
+		if (method.visibility === visibility && node !== undefined && tsModule.isMethodDeclaration(node)) {
+			for (const param of node.parameters) {
+				const name = param.name.getText();
+				const { description, typeHint } = getParameterFromJsDoc(name, method.jsDoc);
+
+				parameters.push({
+					name: name,
+					type: typeToSchemaType(context, typeHint || (param.type != null ? context.checker.getTypeAtLocation(param.type) : undefined)),
+					description: description,
+					optional: param.questionToken !== undefined
+				});
+			}
+
+			// Get return type
+			const signature = context.checker.getSignatureFromDeclaration(node);
+			if (signature != null) {
+				returnType = context.checker.getReturnTypeOfSignature(signature);
+			}
+		}
+
+		// Get return info from jsdoc
+		const { description: returnDescription, typeHint: returnTypeHint } = getReturnFromJsDoc(method.jsDoc);
+
+		yield {
+			kind: "method",
+			name: method.name,
+			privacy: method.visibility,
+			description: method.jsDoc?.description,
+			parameters,
+			return: {
+				description: returnDescription,
+				type: typeToSchemaType(context, returnTypeHint || returnType)
+			},
+			inheritedFrom: getInheritedFromReference(declaration, method, context),
+			summary: getSummaryFromJsDoc(method.jsDoc)
+			// TODO: "static"
+		};
+	}
+}
+
+/**
+ * Returns class member docs for a declaration
+ * @param declaration
+ * @param context
+ */
+function getClassMembersForDeclaration(declaration: ComponentDeclaration, context: TransformerContext): schema.ClassMember[] {
+	return [...getClassFieldsForDeclaration(declaration, context), ...getMethodsForDeclaration(declaration, context)];
```

Comment:
Since you're using generators elsewhere, do you want use one here?

```ts
  yield* getClassFieldsForDeclaration(declaration, context);
  yield* getMethodsForDeclaration(declaration, context);
```